pennylane/transforms/dynamic_one_shot.py

```diff
@@ -287,6 +307,40 @@ def measurement_with_no_shots(measurement):
     return tuple(normalized_meas) if len(normalized_meas) > 1 else normalized_meas[0]
 
 
+def gather_mcm_jit(circuit_measurement, measurement, is_valid):
+    """Combines, gathers and normalizes several measurements with trivial measurement values
+    when the Catalyst compiler is active.
+
+    Args:
+        circuit_measurement (MeasurementProcess): measurement
+        measurement (TensorLike): measurement results
+        samples (List[dict]): Mid-circuit measurement samples
+
+    Returns:
+        TensorLike: The combined measurement outcome
+    """
+    found, meas = False, None
+    for k, meas in measurement.items():
+        if circuit_measurement.mv is k.out_classical_tracers[0]:
+            found = True
+            break
+    if not found:
+        raise LookupError("MCM not found")
+    meas = qml.math.squeeze(meas)
+    if isinstance(circuit_measurement, CountsMP):
+        sum_valid = qml.math.sum(is_valid)
+        count_1 = qml.math.sum(meas * is_valid)
+        return {0: sum_valid - count_1, 1: count_1}
+    if isinstance(circuit_measurement, ProbabilityMP):
+        sum_valid = qml.math.sum(is_valid)
+        count_1 = qml.math.sum(meas * is_valid)
+        counts = qml.math.array(
+            [sum_valid - count_1, count_1], like=qml.math.get_deep_interface(is_valid)
+        )
```

Comment:
Optional suggestion.

```suggestion
    sum_valid = qml.math.sum(is_valid)
    count_1 = qml.math.sum(meas * is_valid)
    if isinstance(circuit_measurement, CountsMP):
        return {0: sum_valid - count_1, 1: count_1}
    if isinstance(circuit_measurement, ProbabilityMP):
        counts = qml.math.array(
            [sum_valid - count_1, count_1], like=qml.math.get_deep_interface(is_valid)
        )
```